### DISCUSSION TITLE
Compute LSR scores for a specific query and document using `compute_query_document_score`

### DISCUSSION THREAD

**Discussion by littlewine** [castorini/pyserini#1386] [Q&A]
Hi all, 

Is it possible to compute individual query-document scores for LSR methods, eg using the `IndexReader.compute_query_document_score` method or some other way? 
When doing retrieval with Anserini, I see that according to the instructions, I have to use `--impact ` and `--pretokenized`, but the IndexReader and its method seem to accept very limited parametrization/arguments (in fact I couldn't even change the default BM25 weights, at least without modifying the framework itself). 

Am I missing something here? 
What would be the best way to achieve this?

**Comment by lintool** [comment]
Hi @littlewine - you're absolutely right... `compute_query_document_score` is currently hard-coded to BM25 with limited parameterization... but we just need to expose the right hooks... PRs welcome!
